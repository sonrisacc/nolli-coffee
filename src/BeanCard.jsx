import React from 'react';
import { string } from 'prop-types';
import styled from 'styled-components';

const Wrapper = styled.div`
  width: 32%;
  border: 2px solid #333;
  border-radius: 4px;
  margin-bottom: 25px;
  padding-right: 10px;
  overflow: hidden;
`;

const Image = styled.img`
  width: 46%;
  float: left;
  margin-right: 10px;
`;

const BeanCard = props => (
  <Wrapper>
    <Image alt={`${props.name} BeanCard`} src={`${props.thumbnail}`} />
    <div>
      <h3>{props.name}</h3>
      <h4>({props.region})</h4>
      <p>{props.description}</p>
    </div>
  </Wrapper>
);

BeanCard.propTypes = {
  name: string.isRequired,
  region: string.isRequired,
  thumbnail: string.isRequired,
  description: string.isRequired,
};

export default BeanCard;
