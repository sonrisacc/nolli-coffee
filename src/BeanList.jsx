import React, { Component } from 'react';
import BeansData from '../exampleBeanData.json';
import BeanCard from './BeanCard';

class BeanList extends Component {
  state = {
    searchTerm: '',
  };

  handleSearchTermChange = event => {
    this.setState({ searchTerm: event.target.value });
  };

  handleFilterBeansInput = () =>
    BeansData.beans.filter(
      cur =>
        `${cur.name} ${cur.region} ${cur.description}`
          .toUpperCase()
          .indexOf(this.state.searchTerm.toUpperCase()) >= 0,
    );

  render() {
    return (
      <div className="beanList">
        <header>
          <h1>Search your beans!</h1>
          <input
            type="text"
            placeholder="Search"
            onChange={this.handleSearchTermChange}
            value={this.state.searchTerm}
          />
        </header>
        <div>
          {this.handleFilterBeansInput().map(curBean => (
            <BeanCard key={curBean.id} {...curBean} />
          ))}
        </div>
      </div>
    );
  }
}
export default BeanList;
