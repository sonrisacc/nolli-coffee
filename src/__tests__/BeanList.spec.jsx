import React from 'react';
import { shallow } from 'enzyme';
// import renderer from 'react-test-renderer';
import BeansData from '../../exampleBeanData.json';
import BeanList from '../BeanList';
import BeanCard from '../BeanCard';

test('BeanList renders correctly', () => {
  // const component = renderer.create(<BeanList />);
  // const tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
  const component = shallow(<BeanList />);
  expect(component).toMatchSnapshot();
});

test('BeanList should render correct amount of beans', () => {
  const component = shallow(<BeanList />);
  expect(component.find(BeanCard).length).toEqual(BeansData.beans.length);
});
