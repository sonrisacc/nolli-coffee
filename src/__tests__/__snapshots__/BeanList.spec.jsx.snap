// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeanList renders correctly 1`] = `
<div
  className="beanList"
>
  <header>
    <h1>
      Search your beans!
    </h1>
    <input
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </header>
  <div>
    <div
      className="sc-bdVaJa jfzRHo"
    >
      <img
        alt="Sumatra BeanCard"
        className="sc-bwzfXH cSKlVC"
        src="https://goo.gl/vxTdPV"
      />
      <div>
        <h3>
          Sumatra
        </h3>
        <h4>
          (
          Indonesia
          )
        </h4>
        <p>
          Dark roast
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa jfzRHo"
    >
      <img
        alt="Costa Rica BeanCard"
        className="sc-bwzfXH cSKlVC"
        src="https://goo.gl/TXRMV5"
      />
      <div>
        <h3>
          Costa Rica
        </h3>
        <h4>
          (
          Costa Rica
          )
        </h4>
        <p>
          Dark roast
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa jfzRHo"
    >
      <img
        alt="Kenya BeanCard"
        className="sc-bwzfXH cSKlVC"
        src="https://goo.gl/vKVnG7"
      />
      <div>
        <h3>
          Kenya
        </h3>
        <h4>
          (
          Indonesia
          )
        </h4>
        <p>
          Dark roast
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa jfzRHo"
    >
      <img
        alt="Kenya BeanCard"
        className="sc-bwzfXH cSKlVC"
        src="https://goo.gl/vKVnG7"
      />
      <div>
        <h3>
          Kenya
        </h3>
        <h4>
          (
          Indonesia
          )
        </h4>
        <p>
          Dark roast
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa jfzRHo"
    >
      <img
        alt="Kenya BeanCard"
        className="sc-bwzfXH cSKlVC"
        src="https://goo.gl/vKVnG7"
      />
      <div>
        <h3>
          Kenya
        </h3>
        <h4>
          (
          Indonesia
          )
        </h4>
        <p>
          Dark roast
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa jfzRHo"
    >
      <img
        alt="Kenya BeanCard"
        className="sc-bwzfXH cSKlVC"
        src="https://goo.gl/vKVnG7"
      />
      <div>
        <h3>
          Kenya
        </h3>
        <h4>
          (
          Indonesia
          )
        </h4>
        <p>
          Dark roast
        </p>
      </div>
    </div>
  </div>
</div>
`;
